git/git.go

```diff
@@ -47,11 +46,12 @@ type CloneRepoOptions struct {
 // be cloned again.
 //
 // The bool returned states whether the repository was cloned or not.
-func CloneRepo(ctx context.Context, opts CloneRepoOptions) (bool, error) {
+func CloneRepo(ctx context.Context, logf func(string, ...any), opts CloneRepoOptions) (bool, error) {
```

Comment:
I think this could be optional as part of options, but not a blocker.

Reply:
I don't think it is optional! If it's not provided, we'll try to log and panic.

Reply:
Oh, it would be via `!= nil` check or assigning a noop function.